language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/VariableDeclarationNodeContext.java

```diff
@@ -60,4 +68,49 @@ public boolean onPreValidation(BallerinaCompletionContext context, VariableDecla
         TextRange equalTokenRange = node.equalsToken().get().textRange();
         return equalTokenRange.endOffset() <= textPosition;
     }
+
+    @Override
+    public void sort(BallerinaCompletionContext context, VariableDeclarationNode node,
+                     List<LSCompletionItem> completionItems) {
+        Optional<TypeSymbol> typeSymbolAtCursor = context.getContextType();
+        if (typeSymbolAtCursor.isEmpty()) {
+            super.sort(context, node, completionItems);
+        }
+
+        TypeSymbol symbol = typeSymbolAtCursor.get();
+        completionItems.forEach(completionItem -> {
+            int rank = SortingUtil.toRank(completionItem, 2);
+
+            if (completionItem.getType() == LSCompletionItem.CompletionItemType.SYMBOL) {
+                SymbolCompletionItem symbolCompletionItem = (SymbolCompletionItem) completionItem;
+                Optional<Symbol> completionSymbol = symbolCompletionItem.getSymbol();
+                if (completionSymbol.isEmpty()) {
+                    completionItem.getCompletionItem().setSortText(SortingUtil.genSortText(rank));
+                    return;
+                }
+                Optional<TypeSymbol> evalTypeSymbol = SymbolUtil.getTypeDescriptor(completionSymbol.get());
+                CompletionItemKind completionItemKind = completionItem.getCompletionItem().getKind();
+                if (evalTypeSymbol.isPresent()) {
+                    switch (completionItemKind) {
```

Comment:
Can't we handle it without considering the item kind?

Reply:
Since we have added the function pointer as an completion item that is of variable kind we require this implementation. So we can't just rely on symbol kind  in this case.

Reply:
What if we add a new CompletionItemType here? Is it possible?

Reply:
IMO, the correct completion item type for function pointer is variable. What is your suggestion as the type?